### PR TITLE
[datadog] DCA 1.12 + Support KSM core basic config

### PR DESCRIPTION
**Pull request by ahmed-mez**
#### What this PR does / why we need it:

- Bump the DCA version
- Support configuring the ksm core check in the cluster agent

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has beed updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/values.yaml [hunk 1]
```diff
@@ -77,6 +77,17 @@ datadog:
     # datadog.kubeStateMetricsNetworkPolicy.create -- If true, create a NetworkPolicy for kube state metrics
     create: false
 
+  # datadog.kubeStateMetricsCore -- Kube-State-Metrics core check configuration
```
Comment (vboulineau) [inline charts/datadog/values.yaml:80]: Only leaf nodes are documented

File: charts/datadog/values.yaml [hunk 2]
```diff
@@ -77,6 +77,13 @@ datadog:
     # datadog.kubeStateMetricsNetworkPolicy.create -- If true, create a NetworkPolicy for kube state metrics
     create: false
 
+  kubeStateMetricsCore:
+    # datadog.kubeStateMetricsCore.enabled -- Enable the kubernetes_state_core check in the Cluster Agent
+    enabled: false
+
+    # datadog.kubeStateMetricsCore.ignoreLegacyKSMCheck -- Disable the auto-configuration of lagacy kubernetes_state check (taken into account only when datadog.kubeStateMetricsCore.enabled is true)
+    ignoreLegacyKSMCheck: true
```
Comment (vboulineau) [inline charts/datadog/values.yaml:85]: Is it necessary to offer this? Setting `datadog.kubeStateMetricsCore.enabled` to false if, for some reason, core does not work well, should be enough no?

Comment (ahmed-mez) [reply charts/datadog/values.yaml:85]: `ignoreLegacyKSMCheck` is only taken into consideration if `kubeStateMetricsCore.enabled=true`. so yes disabling `kubeStateMetricsCore` is enough and guaranteed already.
`ignoreLegacyKSMCheck` can be useful though during the migration phase, users might be interested in enabling both checks at first

File: charts/datadog/values.yaml [hunk 3]
```diff
@@ -77,6 +77,13 @@ datadog:
     # datadog.kubeStateMetricsNetworkPolicy.create -- If true, create a NetworkPolicy for kube state metrics
     create: false
 
+  kubeStateMetricsCore:
+    # datadog.kubeStateMetricsCore.enabled -- Enable the kubernetes_state_core check in the Cluster Agent
```
Comment (clamoriniere) [inline charts/datadog/values.yaml:81]: could you add in comment: require cluster-agent >=1.12 version

Comment (ahmed-mez) [reply charts/datadog/values.yaml]: Done!

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**